crates/compiler/mono/src/borrow.rs

```diff
@@ -232,7 +237,11 @@ impl<'a> ParamMap<'a> {
     fn init_borrow_args(arena: &'a Bump, ps: &'a [(Layout<'a>, Symbol)]) -> &'a [Param<'a>] {
         Vec::from_iter_in(
             ps.iter().map(|(layout, symbol)| Param {
-                borrow: should_borrow_layout(layout),
+                ownership: if should_borrow_layout(layout) {
+                    Ownership::Borrowed
+                } else {
+                    Ownership::Owned
+                },
```

Comment:
this can be cleaned up bit with

```rust
impl Ownership {
    fn from_layout(layout: &Layout) -> Self {
        // inline the implementation of `should_borrow_layout`, then probably remove `should_borrow_layout` itself
    }
}
```

you may need to move `Ownership` to `borrow.rs`, which to me is the more logical place anyway.

Reply:
👍

Reply:
like this?